### PR TITLE
Add other agency type input into form and attribute in database

### LINKED ISSUE #92: Capture "other" agency type description
Related to #75 and #90 

We now restrict the user's selection of agency types to a defined list, but we need to also capture some description from the user when they select the "other" agency type. 

Add an input that's hidden by default and displayed on "other" selection in the agency type dropdown menu.

### PR DESCRIPTION
**Pull request by Nirvikalpa108**
Resolves #92 

### Description
@chaserx and I added a text input field into the partner agency form so that users can describe their agency type if they choose to identify as "Other" from the NDBA agency type list. This is then committed to the database.

### Type of change
* New feature (non-breaking change which adds functionality)

### How Has This Been Tested?
We tested this in the rails console, but need to add a test in the spec directory and will do this shortly.

### CODE REVIEW

File: app/assets/javascripts/partners.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+$(document).on('turbolinks:load', function() {
+    $('form').on('change', '#partner_agency_type', function(event) {
+        console.log("I changed successfully");
+        console.log($(this).val() == "Other");
```
Comment (chaserx) [inline app/assets/javascripts/partners.js]: we can remove these `console.log` statements now

Comment (Nirvikalpa108) [reply app/assets/javascripts/partners.js]: Ha! Thanks @chaserx !!

**Comment by chaserx** [comment]
Hi @seanmarcia we've updated these. Sorry that they were overlooked.

File: Gemfile [hunk 2]
```diff
@@ -55,7 +55,7 @@ group :test do
   gem "chromedriver-helper"
   gem "email_spec"
   gem "selenium-webdriver"
-  gem "shoulda-matchers"
+  gem "shoulda-matchers", "~> 4.0"
```
Comment (benreyn) [inline Gemfile]: My preference would be to leave gems un-version locked, and only lock them when we need to restrict ourselves to certain version.

Comment (chaserx) [reply Gemfile]: sure. carry over from upgrading to avoid deprecation warning there. I'll update.

**Pull request merged**